### PR TITLE
Correctly bookeep threads when we reply on matrix.

### PR DESCRIPTION
**Pull request by Cadair**
replaces #172

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ WORKDIR /usr/src/app
 
 COPY . /usr/src/app
 
-RUN npm install --only=production
+RUN apk add git && npm install --only=production
```
Comment (Half-Shot) [inline Dockerfile]: huh?

File: lib/BridgedRoom.js [hunk 2]
```diff
@@ -412,10 +412,11 @@ BridgedRoom.prototype.onMatrixMessage = async function(message) {
     sendMessageParams.body.username = user.getDisplaynameForRoom(message.room_id);
 
     const reply = await this.findParentReply(message);
+    let parentEvent = undefined;
```
Comment (Half-Shot) [inline lib/BridgedRoom.js]: It's implicit

```suggestion
    let parentEvent;
```

File: lib/BridgedRoom.js [hunk 3]
```diff
@@ -412,10 +412,11 @@ BridgedRoom.prototype.onMatrixMessage = async function(message) {
...
     if (reply !== message.event_id) {
         // We have a reply
-        const parentStoredEvent = await store.getEntryByMatrixId(message.room_id, reply);
-        sendMessageParams.body.thread_ts = parentStoredEvent.remoteEventId;
+        parentEvent = await store.getEntryByMatrixId(message.room_id, reply);
+        sendMessageParams.body.thread_ts = parentEvent.remoteEventId;
```
Comment (Half-Shot) [inline lib/BridgedRoom.js]: What happens if `parentEvent` is undefined?

File: lib/BridgedRoom.js [hunk 4]
```diff
@@ -436,6 +437,16 @@ BridgedRoom.prototype.onMatrixMessage = async function(message) {
         // Add this event to the event store
         const event = new StoredEvent(message.room_id, message.event_id, this._slack_channel_id, res.ts);
         store.upsertEvent(event);
+
+        // If this message is in a slack thread we need to append this message to the end of the thread list.
+        if (parentEvent !== undefined) {
```
Comment (Half-Shot) [inline lib/BridgedRoom.js]:
```suggestion
        if (parentEvent) {
```

File: Dockerfile (hunk 1 above)
Comment (Cadair) [reply Dockerfile]: It needed git to install your slackdown fork :man_shrugging:

File: lib/BridgedRoom.js (hunk 3 above)
Comment (Cadair) [reply lib/BridgedRoom.js]: Then something has gone horribly wrong and we don't have the event that's being replied to?

File: Dockerfile (hunk 1 above)
Comment (Half-Shot) [reply Dockerfile]: Stealing this for develop

File: lib/BridgedRoom.js (hunk 3 above)
Comment (Half-Shot) [reply lib/BridgedRoom.js]: We should definitely give up at that point and throw an error then, we can't assume we have it.

**Pull request closed without merging**